hazelcast/src/main/java/com/hazelcast/map/impl/PartitioningStrategyFactory.java

```diff
@@ -47,17 +51,31 @@ public PartitioningStrategyFactory(ClassLoader configClassLoader) {
      * first attempts locating a {@link PartitioningStrategy} in {code config.getPartitioningStrategy()}. If this is {@code null},
      * then looks up its internal cache of partitioning strategies; if one has already been created for the given
      * {@code mapName}, it is returned, otherwise it is instantiated, cached and returned.
-     * @param mapName       Map for which this partitioning strategy is being created
-     * @param config        the partitioning strategy configuration
+     *
+     * @param mapName                      Map for which this partitioning strategy is being created
+     * @param config                       the partitioning strategy configuration
+     * @param partitioningAttributeConfigs
      * @return
      */
-    public PartitioningStrategy getPartitioningStrategy(String mapName, PartitioningStrategyConfig config) {
+    @SuppressWarnings("checkstyle:NestedIfDepth")
+    public PartitioningStrategy getPartitioningStrategy(
+            String mapName,
+            PartitioningStrategyConfig config,
+            final List<PartitioningAttributeConfig> partitioningAttributeConfigs
+    ) {
         PartitioningStrategy strategy = null;
         if (config != null) {
             strategy = config.getPartitioningStrategy();
             if (strategy == null) {
                 if (cache.containsKey(mapName)) {
                     strategy = cache.get(mapName);
+                } else if (partitioningAttributeConfigs != null && !partitioningAttributeConfigs.isEmpty()) {
+                    if (config.getPartitioningStrategyClass() != null) {
+                        throw new HazelcastException("Partition Strategy Attributes are only supported for built in "
```

Comment:
We throw an exception here, but the documentation says that if there are attributes, we ignore the strategy class.